### ISSUE TITLE
"dev" server not updating

### ISSUE THREAD

**Issue by millertg** [VandyVRC/srophe-eXist-app#212]
@wsalesky @CliffordAnderson @shubenda 
After troubleshooting on 08/05/21, we suspect webhooks from the dev branch are not functioning as they do for the production (aka "master") branch. Consequently, we cannot see what the dev updates look like in html. Once we can see changes made to dev displayed on: dev.architectura.sinica.org, then we can switch both China and US development to dev, reserving prod branch for final publication only. 

Winona, I think you might be the only one who can make this happen--can you help?

**Comment by wsalesky** [comment]
Yes I will take a look.

**Comment by wsalesky** [comment]
@millertg  Looks like the db maybe corrupt, getting a fresh install running and then hopefully everything will run as expected.

**Comment by millertg** [comment]
Ok, great. Thank you so much!

**Comment by millertg** [comment]
@wsalesky I just checked the dev server again and I am not seeing any data. I assume this is because you are still working on it. We thought we should leave the site (dev and prod) alone until you were done, but are worried that the issues are complex. Do you have a sense of when dev might be functioning?

**Comment by wsalesky** [comment]
@millertg  sorry for the delay, I hope to have an update for you tomorrow.

**Comment by millertg** [comment]
@wsalesky ok, thanks!

**Comment by wsalesky** [comment]
@millertg Please take a look now. This should now reflect the development branch for both the application and the data. Let me know if you see any issues, or have trouble with the webhooks.

**Comment by millertg** [comment]
@wsalesky 
It looks good! Thank you!!

**Comment by shubenda** [comment]
Winona,

I have updated k000179 and pushed to the development site: https://github.com/VandyVRC/tcadrt/tree/development/data

But I am getting the: HTTP ERROR 404 Document /db/apps/tcadrt/404.html not found for https://dev.architecturasinica.org/keyword/k000179

When did I do wrong?

Yuh-Fen

**Comment by wsalesky** [comment]
@shubenda can you ty making another change and resubmitting the records? The record on the server has the wrong id (https://architecturasinica.org/keyword/k000173) so there may have been a mix up while I was setting up the dev server. Any minor change should work, even spacing. Then let me know after you submit it.

**Comment by shubenda** [comment]
Winona,

I just made some changes on k000173 and submitted.
Thank you.

Yuh-Fen

**Comment by wsalesky** [comment]
@shubenda Thank you. There is a problem with the webhooks. I will get back to you shortly.

**Comment by shubenda** [comment]
Thank you, Winona!

Not sure if you need this info.
But there was another new keyword doesn't display on dev. Site: k000154

Yuh-Fen

**Comment by wsalesky** [comment]
@shubenda I think this has been fixed. (https://dev.architecturasinica.org/keyword/k000179) Please try again with the other missing keyword, and let me know how it goes.

**Comment by millertg** [comment]
@winona @shubenda I tried to change k000001 from my copy of the dev server, but it shows unresolved conflicts with two records that Yuh Fen changed. These seem up to date in my desktop folder, however. So I am confused.
![Screen Shot 2021-08-10 at 8 59 09 PM](https://user-images.githubusercontent.com/25234059/128958773-84fd067b-e644-4088-9b55-b530318158c4.png)
![Screen Shot 2021-08-10 at 9 05 54 PM](https://user-images.githubusercontent.com/25234059/128958790-252a3aa8-14f2-46f5-9018-ad6198a66d4b.png)
![Screen Shot 2021-08-10 at 9 09 17 PM](https://user-images.githubusercontent.com/25234059/128959036-e7b2858f-8370-4b59-b070-15f6aaf25ca4.png)

**Comment by wsalesky** [comment]
@millertg I'm not sure what your process is, but it is very important to do a pull request on the development branch before you start working, or before you commit your work. This will help eliminate merge conflicts. As it is you will need to resolve the current conflicts before you can merge your branch into the development branch. I use the tool mergeTool, but perhaps you can also use the GitHub desktop tool to resolve the conflicts? Another option is to checkout the development branch,  do a pull request to make sure it is up to date, create your branch, edit k000001 and then do a new pull request. You may also want to get @CliffordAnderson  to walk you through one merge conflict in person, just so you have an idea how to resolve conflicts, as they will arise occasionally. 

Only after you have checked your changes on the dev server will you want to merge them into the production server/main branch.

**Comment by millertg** [comment]
@wsalesky @shubenda I figured out how to fix it (whew!), but probably in the most time-consuming way possible.... 
Also was able to comment out in-process text in Dev and push to Prod, do that works too. 
Now, we have 154 and 179 in Prod. 
Yuh-Fen--will you update your Dev and Prod, then add something in Dev and push to Prod? I want to make sure we can both update both without running into conflicts...
Thanks for your patience!

**Comment by millertg** [comment]
and perseverance ;-)

**Comment by millertg** [comment]
@wsalesky @shubenda 
We merged the master back in to development. The result seemed good, but now, although I see the files for both k000179 and k000154 in dev and prod, those two are not displaying on the dev server (404 Error). Could we be back to a webhook issue?

**Comment by wsalesky** [comment]
I will investigate.
-Winona

**Comment by CliffordAnderson** [comment]
I am happy to assist with handling merge conflicts, but it sounds like @millertg and @shubenda have things under control now.

**Comment by millertg** [comment]
@CliffordAnderson Thanks for the offer--we may yet have need to take you up on that!

**Comment by wsalesky** [comment]
@millertg I am working on webhooks this afternoon, if you can not push any new changes while I test that would be great!

**Comment by millertg** [comment]
@wsalesky @shubenda 
Of course! We will stay away from the site until you give us the word.

**Comment by wsalesky** [comment]
@millertg @shubenda Okay, we should be good.

**Comment by millertg** [comment]
@wsalesky Ok, I can see something to update in 179, so I will give it a try

**Comment by millertg** [comment]
@wsalesky @shubenda I changed a little bit of the text in k000195. It seems to work very well! Except dev seems to be running an old version of the Zotero library. Can we make it so that the dev server pulls from the most updated version of the Zotero library? Or will we need to run a separate update for dev on Zotero?

**This is from the Dev server:** 
![Screen Shot 2021-08-13 at 4 06 30 PM](https://user-images.githubusercontent.com/25234059/129421451-4f336a6f-0976-4b64-8e76-4faa5646753e.png)

**This is from the Prod server:** 
![Screen Shot 2021-08-13 at 4 45 50 PM](https://user-images.githubusercontent.com/25234059/129421473-a994b4fc-6717-4618-a7d2-d4fcd9999efe.png)

**Comment by wsalesky** [comment]
@millertg I updated it for you, but you will need to run periodic updates just like on the production site.

**Comment by millertg** [comment]
@wsalesky Ok, good--that is what I thought! Would the update url for the development site then be: https://dev.architecturasinica.org/apps/zotero2bibl/indes.html

**Comment by millertg** [comment]
@wsalesky The dev Zotero update works! And only took 9 seconds to complete an update. Thank you!!
